### PR TITLE
Refactor validation logic

### PR DESCRIPTION
**Pull request by DmitriyMolch**
Made health service validation to return multiple validation errors

### CODE REVIEW

File: api/src/services/healthService.js [hunk 1]
```diff
@@ -8,26 +8,21 @@ const SEPARATOR = ',';
...
+  
+  let validationErrors = [
+    ...validateBlockchains(blockchains),
+    ...validateNetworks(networks),
+    ...validateGroups(groups),
+    ...validateAddresses(addresses, accounts)
+  ];
+
+  if (validationErrors.length) {
+    return res.status(400).json({ msg: validationErrors });
```
Comment (AndreiNavoichyk) [inline api/src/services/healthService.js]: `errors` instead of `msg`

File: api/src/services/healthService.js [hunk 2]
```diff
@@ -51,140 +46,83 @@ async function checkAccountsBalances(req, res) {
...
-    validateNetworksNames(networks);
-  } catch (error) {
-    return res.status(400).json({
-      msg: error.message,
-    });
+  
+  if (networks) {
+    const validationErrors = validateNetworks(networks);
+    if (validationErrors.length) {
+      return res.status(400).json({msg: validationErrors });
```
Comment (AndreiNavoichyk) [inline api/src/services/healthService.js:119]: errors

File: api/src/services/healthService.js [hunk 3]
```diff
@@ -203,19 +141,20 @@ async function checkBlockchainBlocksFinalization(req, res) {
...
+    }
+  } else {
+    networks = Object.values(networkNames);
```
Comment (AndreiNavoichyk) [inline api/src/services/healthService.js]: It should be required?

Comment (DmitriyMolch) [reply api/src/services/healthService.js]: No, it's simpler not to have it required in terms API URL usage.

File: api/src/services/healthService.js (hunk 1 above)
Comment (DmitriyMolch) [reply api/src/services/healthService.js]: Last time we talked that in other places of the code we have `msg` for validation errors. Do we want to change it?

Comment (AndreiNavoichyk) [reply api/src/services/healthService.js]: msg -> "Validation failed"
errors -> list of errors

File: api/src/services/healthService.js [hunk 4]
```diff
@@ -51,140 +46,83 @@ async function checkAccountsBalances(req, res) {
...
-      try {
-        await cereNetworkService.getHealth(network);
-      } catch (error) {
-        console.error(error);
-        errors.push(`Unable to connect to ${network} network`);
-      }
-    }));
-
-    res.status(200).json(errors.length ? { errors } : { msg: `${networks.join(',')} status is ok`});
+  res.status(200).json(errors.length ? { errors } : { msg: `${networks.join(',')} status is ok`});
```
Comment (AndreiNavoichyk) [inline api/src/services/healthService.js]: msg -> "X failed"
errors -> list of errors

File: api/src/services/healthService.js (hunk 2 above)
Comment (AndreiNavoichyk) [inline api/src/services/healthService.js]: msg -> "X failed"
errors -> list of errors

File: api/src/services/healthService.js [hunk 5]
```diff
@@ -203,19 +141,20 @@ async function checkBlockchainBlocksFinalization(req, res) {
...
-    validateNetworksNames(networks);
-  } catch (error) {
-    return res.status(400).json({
-      msg: error.message,
-    });
+  
+  if (networks) {
+    const validationErrors = validateNetworks(networks);
+    if (validationErrors.length) {
+      return res.status(400).json({msg: validationErrors });
```
Comment (AndreiNavoichyk) [inline api/src/services/healthService.js]: msg -> "X failed"
errors -> list of errors

File: api/src/services/healthService.js (hunk 3 above)
Comment (AndreiNavoichyk) [reply api/src/services/healthService.js]: Check spec, please

**Review by AndreiNavoichyk — COMMENTED** [review]
Check spec

File: api/src/services/healthService.js (hunk 1 above)
Comment (DmitriyMolch) [reply api/src/services/healthService.js]: fixed

File: api/src/services/healthService.js (hunk 3 above)
Comment (DmitriyMolch) [reply api/src/services/healthService.js]: fixed

File: api/src/services/healthService.js (hunk 4 above)
Comment (DmitriyMolch) [reply api/src/services/healthService.js]: fixed

File: api/src/services/healthService.js (hunk 2 above)
Comment (DmitriyMolch) [reply api/src/services/healthService.js]: fixed

File: api/src/services/healthService.js (hunk 5 above)
Comment (DmitriyMolch) [reply api/src/services/healthService.js]: fixed

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**